src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -685,6 +685,17 @@ internal void InsertAttributeExpensive(int insertAtIndex, int sequence, string a
         public ArrayRange<RenderTreeFrame> GetFrames() =>
             _entries.ToRange();
 
+        internal void AssertTreeIsValid(IComponent component)
+        {
+            if (_openElementIndices.Count > 0)
+            {
+                // It's never valid to leave an element/component/region unclosed. Doing so
+                // could cause undefined behavior in diffing.
+                ref var invalidFrame = ref _entries.Buffer[_openElementIndices.Peek()];
```

Comment:
Is it always guaranteed that the index is going to be in range (assuming there are no bugs in the RenderTreeBuilder code) I believe that's the case, but given we are dealing with "failure modes" here I ask just in case to be more cautious.

Reply:
Yes, but even if not, the code here is certain to throw (either from the `throw` statement on L695, or an index-out-of-range on L694), which is all we need.